Framework/CurveFitting/src/IMWDomainCreator.cpp

```diff
@@ -145,45 +145,42 @@ std::pair<size_t, size_t> IMWDomainCreator::getXInterval() const {
 
   setParameters();
 
-  // find the fitting interval: from -> to
+  // From points to the first occurrence of StartX in the workspace interval.
+  // End points to the last occurrence of EndX in the workspace interval.
```

Comment:
I don't think these two lines are totally correct. "from" is an iterator pointing to the first value in the X vector that falls into the interval. "to" is an iterator to the last X value in the interval.